riptide-core/src/test/java/org/zalando/riptide/IOTest.java

```diff
@@ -92,10 +96,23 @@ public void shouldReadContributors() throws IOException, ExecutionException, Int
     }
 
     @Test
-    public void shouldCancelRequest() throws IOException, ExecutionException, InterruptedException {
-        rest.get("/repos/{org}/{repo}/contributors", "zalando", "riptide")
+    public void shouldCancelRequest() throws ExecutionException, InterruptedException {
+        driver.addExpectation(onRequestTo("/repos/zalando/riptide/contributors"),
+                giveEmptyResponse().after(1, TimeUnit.SECONDS));
+
+        final CompletableFuture<Void> future = rest.get("/repos/{org}/{repo}/contributors", "zalando", "riptide")
                 .dispatch(series(),
-                        on(SUCCESSFUL).call(pass())).cancel(true);
+                        on(SUCCESSFUL).call(pass()));
+
+        future.cancel(true);
+
+        try {
+            future.join();
+        } catch (final CancellationException e) {
+            // expected
+        }
+
+        driver.reset();
```

Comment:
I can only see and `executor.shutdown()` ...

Anyhow, why do you need to reset at all? According to the theory you mentioned last week, it should run in an isolated test instance.